### PR TITLE
Addressing Windows Path issue causing unit test_automated_training.py unit tests to fail

### LINKED ISSUE #604: Add compatibility for Windows
We would like to add compatibility for `Windows` in `ivadomed`. Currently we are only `Linux` and `Mac` compatible.

### Motivation for the feature
In this review, some of the reviewers mentioned that they had issues trying to install `ivadomed` on `Windows`: https://github.com/openjournals/joss-reviews/issues/2868#issuecomment-752687437

### Description of the feature
To address the issue in the review, we will add to our documentation that we do not currently support Windows. However, it would be nice to support Windows in the future.

### PR DESCRIPTION
**Pull request by dyt811**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [X] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [X] I've assigned a reviewer

#### PR contents

- [X] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [X] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

TLDR: fixed Windows pathing related bug that causes prediction files not to be generated at the correct location (stuck in the old tmp location)

Identified the root cause of #604 for unit test output issues which prevented adding Windows CI to GitHub Action (that is coming soon). 
- Applied Pathlib approach to avoid manual str parsing. 
- This will no longer result in malformed file path on Windows
- _**Prior to this fix, no Windows users can generate test prediction in the proper location**_ (it will be left at the old source input folder... surprising we have not heard a single complaint from end users about this yet. 

## How to test this PR:
- Be on windows. 
- Pull down this PR branch and ensure all other Python ENV is setup. 
- Making SURE to `pip uninstall ivadomed` and then reinstall it via ` pip install -e .`
- Ensuring the test files have been downloaded
- Run pytest from repo root folder. 

## Linked issues

Partially resolve #604 by unlocking GitHub Actions to be setup against Windows Environment as well going forward for all PRs and future automated CIs, first step towards ~world domination~ Windows 10 cross compatibility

### CODE REVIEW

**Comment by dyt811** [comment]
@charleygros could you briefly review this very small PR? It addresses some important issue discovered preventing future wider Windows testing while the correction is very small.

**Comment by dyt811** [comment]
@charleygros could you briefly review this very small PR? It addresses some important issue discovered preventing future wider Windows testing while the correction is very small.

**Review by charleygros — APPROVED** [review]
🥇
